src/lib/markbind/src/parser.js

```diff
@@ -197,193 +178,19 @@ class Parser {
       // Render inner file content
       const { content: renderedContent, childContext, userDefinedVariables }
         = this._renderIncludeFile(filePath, element, context, config);
-      if (!Parser.PROCESSED_INNER_VARIABLES.has(filePath)) {
-        Parser.PROCESSED_INNER_VARIABLES.add(filePath);
-        this._extractInnerVariables(renderedContent, childContext, config);
-      }
+      this.extractInnerVariablesIfNotProcessed(renderedContent, childContext, config, filePath);
       const innerVariables = this.getImportedVariableMap(filePath);
       Parser.VARIABLE_LOOKUP.get(filePath).forEach((value, variableName, map) => {
         map.set(variableName, nunjucks.renderString(value, { ...userDefinedVariables, ...innerVariables }));
       });
     });
   }
 
-  _preprocess(node, context, config) {
-    const element = node;
-    const self = this;
-    element.attribs = element.attribs || {};
-    element.attribs[ATTRIB_CWF] = path.resolve(context.cwf);
-    if (element.name === 'thumbnail') {
-      return Parser._preprocessThumbnails(element);
-    }
-    const requiresSrc = ['include'].includes(element.name);
-    if (requiresSrc && _.isEmpty(element.attribs.src)) {
-      const error = new Error(`Empty src attribute in ${element.name} in: ${element.attribs[ATTRIB_CWF]}`);
-      this._onError(error);
-      return Parser.createErrorNode(element, error);
-    }
-    const shouldProcessSrc = ['include', 'panel'].includes(element.name);
-    const hasSrc = _.hasIn(element.attribs, 'src');
-    let isUrl;
-    let includeSrc;
-    let filePath;
-    let actualFilePath;
-    if (hasSrc && shouldProcessSrc) {
-      isUrl = utils.isUrl(element.attribs.src);
-      includeSrc = url.parse(element.attribs.src);
-      filePath = isUrl
-        ? element.attribs.src
-        : path.resolve(path.dirname(context.cwf), decodeURIComponent(includeSrc.path));
-      actualFilePath = filePath;
-      const isBoilerplate = _.hasIn(element.attribs, 'boilerplate');
-      if (isBoilerplate) {
-        element.attribs.boilerplate = element.attribs.boilerplate || path.basename(filePath);
-        actualFilePath
-          = Parser.calculateBoilerplateFilePath(element.attribs.boilerplate, filePath, config);
-        this.boilerplateIncludeSrc.push({ from: context.cwf, to: actualFilePath });
```

Comment:
Is `boilerplateIncludeSrc` necessary?
For both panels and includes we eventually still push either `dynamicIncludeSrc` or `staticIncludeSrc`